modules/message-broker-api/src/main/java/org/opencastproject/message/broker/api/assetmanager/AssetManagerItem.java

```diff
@@ -292,17 +294,16 @@ public String apply(DeleteEpisode a) {
    * @return Builds a {@link AssetManagerItem} for taking a media package snapshot.
    */
   public static TakeSnapshot add(Workspace workspace, MediaPackage mp, AccessControlList acl, long version, Date date) {
-    String dc = null;
-    Opt<DublinCoreCatalog> episodeDublincore = DublinCoreUtil.loadEpisodeDublinCore(workspace, mp);
-    if (episodeDublincore.isSome()) {
-      try {
-        dc = episodeDublincore.get().toXmlString();
-      } catch (IOException e) {
-        throw new IllegalStateException(
-                String.format("Not able to serialize the episode dublincore catalog %s.", episodeDublincore.get()), e);
+    String dcXml = null;
+    for (Catalog catalog: mp.getCatalogs(MediaPackageElements.EPISODE)) {
+      try (InputStream in = workspace.read(catalog.getURI())) {
+        dcXml = IOUtils.toString(in, StandardCharsets.UTF_8);
+      } catch (Exception e) {
+        throw new IllegalStateException(String.format("Unable to load dublin core catalog for event '%s'",
+                mp.getIdentifier()), e);
       }
     }
-    return new TakeSnapshot(mp.getIdentifier().compact(), MediaPackageParser.getAsXml(mp), dc,
+    return new TakeSnapshot(mp.getIdentifier().compact(), MediaPackageParser.getAsXml(mp), dcXml,
```

Comment:
@lkiesow The semantic difference is that the former could would have failed early in case of an invalid XML while the new code will not check anything considering that data it loads from the workspace but simply pass the content to TakeSnapshot. 

Have you considered possible impacts this could have?

Reply:
yes